virttest/utils_misc.py

```diff
@@ -840,6 +840,12 @@ def archive_as_tarball(source_dir, dest_dir, tarball_name=None,
     >>> utils_misc.archive_as_tarball('/tmp', '/net/server/backup',
                                       'host1-tmp', 'gz')
     '''
+    # Refer to https://docs.python.org/3/library/tarfile.html#tarfile.open
+    supported_compression = ['gz', 'bz2', 'xz']
+    if compression not in supported_compression:
+        raise exceptions.TestError("compression %s is not supported method %s"
+                                   % (compression, supported_compression))
+
```

Comment:
xz (lzma) is not really supported on Python 2.6 and 2.7:

```
tarfile.open('tarball.tar.xz', 'r:xz')
---------------------------------------------------------------------------
CompressionError                          Traceback (most recent call last)
<ipython-input-2-4bd68a104a27> in <module>()
----> 1 f = tarfile.open('tarball.tar.xz', 'r:xz')

/usr/lib64/python2.7/tarfile.pyc in open(cls, name, mode, fileobj, bufsize, **kwargs)
   1682                 func = getattr(cls, cls.OPEN_METH[comptype])
   1683             else:
-> 1684                 raise CompressionError("unknown compression type %r" % comptype)
   1685             return func(name, filemode, fileobj, **kwargs)
   1686 

CompressionError: unknown compression type 'xz'
```

Please refer to: https://docs.python.org/2.6/library/tarfile.html#tarfile.open